modules/core/src/main/kotlin/datamaintain/core/step/sort/ByCaseInsensitiveSeparatorFreeAlphabeticalSortingStrategy.kt

```diff
@@ -1,65 +1,55 @@
 package datamaintain.core.step.sort
 
 import datamaintain.core.script.Script
+import java.util.*
 
 /**
  * Allow to sort scripts considering the parts of the value returned by the getter.
  * The parts list is given by a split on all chars that are not a figure or a letter.
  * The number parts would be sorted as number, and the others as string (so with alphabetical sort)
  *
- * We assume the differents values have an homogeneity. For instance, you will have troubles
+ * We assume the different values have a homogeneity. For instance, you will have troubles
  * to sort ["abc.123", "abc.234", "abc.0001", "abc.def"]
  */
 class ByCaseInsensitiveSeparatorFreeAlphabeticalSortingStrategy : SortingStrategy<String>() {
     private val numberRegex = "^\\d*$".toRegex()
 
     override fun <T : Script> sort(scripts: List<T>, getter: (T) -> String): List<T> {
-        return scripts.sortedWith(Comparator { script1, script2 ->
+        return scripts.sortedWith { script1, script2 ->
             val value1 = split(getter(script1))
             val value2 = split(getter(script2))
 
-            var compareResult = 0;
+            val compareResult: Int
 
-            for ((index, part1) in value1.withIndex()) {
-                if (value2.size <= index) {
-                    // value1 is after value2 because value1 has more parts than value2
-                    compareResult = 1;
-                    break
+            //Go to first difference or until the end of shorter value
+            var index = 0
+            while (index < value1.size && index < value2.size && value1[index] == value2[index]) {
```

Comment:
Est ce que ce n'est pas plus simple de faire une itération sur le plus petit des 'value' et directement chercher une différence plutot que de toujours vérifier la taille et d'incrémenter un index ?
```
val minLength = if (value1.length < value2.length) value1.length else value2.length
val lastIndex: Int? = (0 until minLength).dropWhile { value1[it] == value2[it] }.firstOrNull()
```


Pour compareResult : Si lastIndex a une valeur alors il y a une différence entre les deux chaînes alors que null signifie que toute la chaine est égale sur la partie comparé